proxy/src/main.rs

```diff
@@ -36,23 +36,6 @@ async fn flatten_err(
     f.map(|r| r.context("join error").and_then(|x| x)).await
 }
 
-/// A proper parser for auth backend parameter.
-impl clap::ValueEnum for auth::BackendType<()> {
```

Comment:
Was nice to have that in one place instead of specifying manually list of variants and match around them. I understand that it's no longer that simple because of new parameters in backend type variants. Should be possible to solve with additional BackendTypeKind (name is already awkward), but probably it would cause only mental overhead without any benefit, so likely it is not worth it

Reply:
Yeah, I don't think it's worth it, especially since I grew tired of current parser impl and will likely rewrite it using something derive-based.

Reply:
Furthermore, I don't believe that verbose impl was better anyway. In a hindsight, the proposed way to define variants by returning a slice of allowed names doesn't look definitively superior. Especially considering the fact that I immediately had to drop it once things got a bit trickier.